products/bash/setup-demos.sh

```diff
@@ -313,27 +362,71 @@ done
 #-------------------------------------------------------------------------------------------------------------------
 # Add the required addons
 #-------------------------------------------------------------------------------------------------------------------
-$DEBUG && divider
-$DEBUG && echo "Addons:"
+$DEBUG && divider && echo "Addons:"
 for row in $(echo "${REQUIRED_ADDONS_JSON}" | jq -r '.[] | select(.enabled == true ) | @base64'); do
   ADDON_JSON=$(echo ${row} | base64 --decode)
   $DEBUG && echo $ADDON_JSON | jq .
 done
 
 #-------------------------------------------------------------------------------------------------------------------
-# Add the required namespaces
+# Display the required namespaces and create secrets and additional namespaces if does not exist
 #-------------------------------------------------------------------------------------------------------------------
-$DEBUG && divider
-$DEBUG && echo "Namespaces:"
-for namespace in $(echo "${REQUIRED_PRODUCTS_JSON}" | jq -r '[ .[] | select(.enabled == true ) | .namespace ] | unique | .[]'); do
-  $DEBUG && echo $namespace
+$DEBUG && divider && echo "Namespaces:"
+
+# add main namespace to status if exists
+oc get project $NAMESPACE 2>&1 >/dev/null
+if [ $? -ne 0 ]; then
+  $DEBUG && echo -e "\n$cross [ERROR] Namespace '$NAMESPACE' does not exist"
+  update_status "Namespace '$NAMESPACE' does not exist" "namespace" "$FAILURE_CODE" "$($GET_UTC_TIME)" "Getting" "Failed" "$NAMESPACE"
+else
+  $DEBUG && echo -e "\n$tick [SUCCESS] Namespace '$NAMESPACE' already exists"
+  update_status "Namespace '$NAMESPACE' already exists" "namespace" "$SUCCESS_CODE" "$($GET_UTC_TIME)" "Getting" "Pending" "$NAMESPACE"
```

Comment:
Note [this from the playbook](https://github.ibm.com/cp4i/cp4i-playbook/blob/master/software/operators/CRDs-Operand-Status.md#when-to-create-a-condition):
> In general, Conditions should be used to describe abnormal conditions.

So I think only add conditions for errors, not for success